RTClib.cpp

```diff
@@ -746,6 +746,168 @@ TimeSpan::TimeSpan(int16_t days, int8_t hours, int8_t minutes, int8_t seconds)
 /**************************************************************************/
 TimeSpan::TimeSpan(const TimeSpan &copy) : _seconds(copy._seconds) {}
 
+/*!
+    @brief  Create a new TimeSpan from an iso8601 formatted period string
+
+    If the string is entirely malformed (doesn't begin with P or -) this will
+    construct a TimeSpan of 0 seconds. If parsing fails before the end of the
+    string, this constuctor will interpret its argument as the longest
+    well-formed prefix string.
+
+    For example, the string P5M10~ will parse the same as P5M
+
+    @param iso8601 the formatted string
+*/
+TimeSpan::TimeSpan(char *iso8601) : _seconds(0) {
```

Comment:
`iso8601` and `cursor` should be `const char *`

Reply:
In an earlier revision (before my first commit to this PR) they were. The problem is that `strtol`'s endptr argument is of type `char**`, not `char const**`, so later assigning `rest` to `cursor` is not allowed (Arduino builds with -fpermissive, so it's a warning not an error).

Reply:
Are you sure it's not allowed? Assignments that remove `const` are not allowed but, if I am not mistaken, adding `const` is OK. According to [cppreference.com](https://en.cppreference.com/w/cpp/language/cv), “reference/pointer to _unqualified_ type can be converted to reference/pointer to `const`”. Also, avr-g++ 7.3.0 (as distributed with Arduino 1.8.13) doesn't emit warnings on this, even with `-Wall -Wextra`.

Reply:
I misspoke about the exact nature of the problem. If I make `iso8601`, `cursor` and `rest` `const char*`, the exact error message is
```
RTClib.cpp:803:26: warning: invalid conversion from 'const char**' to 'char**' [-fpermissive]
num = strtol(cursor, &rest, 10);
```

Reply:
Oh, I see. The thing is, `rest` should remain `char *`, without `const`.